File: src/main/java/com/domclick/dto/request/AccountTransferRequest.java
package com.domclick.dto.request;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.PositiveOrZero;
import java.io.Serializable;
import java.math.BigDecimal;

public class AccountTransferRequest implements Serializable {
    @NotNull(message = "Field 'fromAccountId' is required")
    @PositiveOrZero(message = "Field 'fromAccountId' value must be positive")
    private Long fromAccountId;
    @NotNull(message = "Field 'toAccountId' is required")
    @PositiveOrZero(message = "Field 'toAccountId' value must be positive")
    private Long toAccountId;
    @NotNull(message = "Field 'value' is required")
    @PositiveOrZero(message = "Field 'value' value must be positive")
    private BigDecimal value;

    public Long getFromAccountId() {
        return fromAccountId;
    }

    public void setFromAccountId(Long fromAccountId) {
        this.fromAccountId = fromAccountId;
    }

    public Long getToAccountId() {
        return toAccountId;
    }

    public void setToAccountId(Long toAccountId) {
        this.toAccountId = toAccountId;
    }

    public BigDecimal getValue() {
        return value;
    }

    public void setValue(BigDecimal value) {
        this.value = value;
    }
}

File: app/src/main/kotlin/com/domclick/dto/request/AccountTransferRequest.kt
package com.domclick.dto.request

import javax.validation.constraints.NotNull
import javax.validation.constraints.PositiveOrZero
import java.io.Serializable
import java.math.BigDecimal

class AccountTransferRequest : Serializable {
    @NotNull(message = "Field 'fromAccountId' is required")
    @PositiveOrZero(message = "Field 'fromAccountId' value must be positive")
    var fromAccountId: Long = 0
    @NotNull(message = "Field 'toAccountId' is required")
    @PositiveOrZero(message = "Field 'toAccountId' value must be positive")
    var toAccountId: Long = 0
    @NotNull(message = "Field 'value' is required")
    @PositiveOrZero(message = "Field 'value' value must be positive")
    lateinit var value: BigDecimal
}
